kuksa_databroker/databroker/src/types.rs

```diff
@@ -390,6 +390,20 @@ impl DataValue {
         }
     }
 
+    pub fn find(&self, search_value: &DataValue) -> Result<bool, CastError> {
```

Comment:
And I think `contains` would be a better name for this function (`find` is usually used when the found element or it's index is returned by the function).

But since the functionality is only applicable to `DataValue`s that are arrays it could make more sense to not have this as a general function for `DataValue`s at all, but instead move the functionality directly into `validate_allowed` (as that's probably the only place it will be used anyway).

If keeping it as a generic function, a new error type should probably be created with `WrongType` and `NotAnArray` as possible error kinds, since CastError isn't that fitting (I think).

Reply:
`find` can be removed as it's no longer used.